hail/src/main/scala/is/hail/HailContext.scala

```diff
@@ -76,16 +76,15 @@ object HailContext {
     val javaVersion = raw"(\d+)\.(\d+)\.(\d+).*".r
     val versionString = System.getProperty("java.version")
     versionString match {
-      // old-style version: 1.MAJOR.MINOR
-      // new-style version: MAJOR.MINOR.SECURITY (started in JRE 9)
+      // old-style version: 1.MAJOR.MINOR (before JRE 9)
+      // new-style version: MAJOR.MINOR.SECURITY (starting with JRE 9)
       /* see:
        * https://docs.oracle.com/javase/9/migrate/toc.htm#JSMIG-GUID-3A71ECEF-5FC5-46FE-9BA9-88CBFCE828CB */
-      case javaVersion("1", major, _) =>
-        if (major.toInt < 8)
-          fatal(s"Hail requires Java 1.8, found $versionString")
+      case javaVersion("1", _, _) =>
+        warn(s"Hail is tested against Java 11, found Java $versionString")
       case javaVersion(major, _, _) =>
         if (major.toInt != 11)
-          fatal(s"Hail requires Java 8 or 11, found $versionString")
+          warn(s"Hail is tested against Java 11, found $versionString")
```

Comment:
I don't fully grok what we were trying to accomplish here.

You can't possibly get to these lines if the JVM version is lower than the byte code, so JVMs <=7 would never execute this. AFAICT, Spark now supports all LTS versions of Java (currently: 8, 11, 17), so I see no strong reason to prohibit any version. Instead, I simply warn if you are using a version of the JVM other than the one with which we test.

Reply:
In particular:

> Spark runs on Java 8/11/17, Scala 2.12/2.13, Python 3.8+, and R 3.5+. Java 8 prior to version 8u371 support is deprecated as of Spark 3.5.0.

https://spark.apache.org/docs/latest/#downloading